core/src/main/scala/cats/syntax/apply.scala

```diff
@@ -44,20 +41,209 @@ private[syntax] trait ApplySyntaxBinCompat0 {
     new IfApplyOps[F](fa)
 }
 
-final class ApplyBinCompat1[F[_], A](private val fa: F[A]) extends AnyVal {
-  def ap[B, C](fb: F[B])(implicit ev$1: A <:< (B => C), F: Apply[F]): F[C] =
-    F.ap(fa.asInstanceOf[F[B => C]])(fb)
-  def productR[B](fb: F[B])(implicit F: Apply[F]): F[B] = F.productR[A, B](fa)(fb)
-  def productL[B](fb: F[B])(implicit F: Apply[F]): F[A] = F.productL[A, B](fa)(fb)
-  @inline final def <*>[B, C](fa: F[B])(implicit ev$1: A <:< (B => C), F: Apply[F]): F[C] =
-    F.<*>[B, C](fa.asInstanceOf[F[B => C]])(fa)
-  @inline def *>[B](fb: F[B])(implicit F: Apply[F]): F[B] = F.*>[A, B](fa)(fb)
-  @inline def <*[B](fb: F[B])(implicit F: Apply[F]): F[A] = F.<*[A, B](fa)(fb)
-  def ap2[B, C, D](fa: F[B], fb: F[C])(implicit ev$1: A <:< ((B, C) => D), F: Apply[F]): F[D] =
-    F.ap2[B, C, D](fa.asInstanceOf[F[(B, C) => D]])(fa, fb)
-  def map2[B, C](fb: F[B])(f: (A, B) => C)(implicit F: Apply[F]): F[C] = F.map2[A, B, C](fa, fb)(f)
-  def map2Eval[B, C](fb: Eval[F[B]])(f: (A, B) => C)(implicit F: Apply[F]): Eval[F[C]] =
-    F.map2Eval[A, B, C](fa, fb)(f)
+private[syntax] trait ApplySyntaxBinCompat1 {
+  implicit final def catsSyntaxApplyFABOps[F[_], A, B](fab: F[A => B]): ApplyFABOps[F, A, B] =
+    new ApplyFABOps[F, A, B](fab)
+
+  implicit final def catsSyntaxApply2Ops[F[_], A, B, C](ff: F[(A, B) => C]): Apply2Ops[F, A, B, C] =
+    new Apply2Ops[F, A, B, C](ff)
+
+  implicit final def catsSyntaxProductOps[F[_], A, B](fa: F[A]): ProductOps[F, A, B] =
+    new ProductOps[F, A, B](fa)
+
+  implicit final def catsSyntaxMap2Ops[F[_], A, B, C](fa: F[A]): Map2Ops[F, A, B, C] =
+    new Map2Ops[F, A, B, C](fa)
```

Comment:
Why separate these ones?

Reply:
There is no strong reason to separate them, so I can put these together. (but it was a bit difficult to come up with a unique syntax name that doesn't conflict with exisisting one. `ApplySyntaxFA` or `ApplyOps2` would be fine?)

Reply:
`ApplyOps2` or `ApplyOpsBincompat1` or similar seems fine. We probably have some similar situations somewhere else in Cats.

Reply:
Fix: [f97affd](https://github.com/typelevel/cats/pull/4477/commits/f97affd283ed04481b9bb4a5c97559f72099882e)